spec/models/course_spec.rb

```diff
@@ -142,9 +142,457 @@
+          end
+          it 'should not create a new assignment' do
+            # check if the number of assignments stored has changed
+            expect(test_course.assignments.length).to eq(1)
+          end
+        end
+        context 'when the assignment is new to the database' do
+          it 'should set assignment_properties.repository_folder, token_period and ' \
+             'unlimited_tokens to pre-determined values' do
+            # setting properties to different values than the ones they are supposed to be changed to
+            @assignment_before_call.token_period = nil
+            @assignment_before_call.unlimited_tokens = nil
+
+            # desired list of assignment attributes
+            desired_attributes = Assignment::DEFAULT_FIELDS.map do |f|
+              @assignment_before_call.public_send(f)
+            end
+
+            # this is done to assure assignment_properties.repository_folder = row[0]
+            desired_attributes[0] = 'TEST_FOLDER'
+
+            # csv representation of the single assignment
+            csv = desired_attributes.to_csv
+
+            test_course.upload_assignment_list('csv', csv)
+
+            assignment_after_call = test_course.assignments.find_by(short_identifier: 'TEST_FOLDER')
+
+            expect(assignment_after_call.assignment_properties.repository_folder).to eq('TEST_FOLDER')
+            expect(assignment_after_call.assignment_properties.token_period).to eq(1)
+            expect(assignment_after_call.assignment_properties.unlimited_tokens).to eq(false)
+          end
+          it 'should save the new object to the database with the intended attributes' do
+            desired_attributes = ['short_identifier', 'description', 1.day.from_now.at_beginning_of_minute, 'message',
+                                  1, 2, 1, true, true, 2.days.from_now.at_beginning_of_minute, 'remark_message',
+                                  true, true, false, true, true, true, true, false, true, true]
+            csv = desired_attributes.to_csv
+
+            test_course.upload_assignment_list('csv', csv)
+
+            assignment = test_course.assignments.find_by(short_identifier: 'short_identifier')
+
+            # Check that the assignment is saved in the database
+            expect(assignment).not_to eq(nil)
+
+            # Check that al the desired attributes match those stored in the assignment
+            Assignment::DEFAULT_FIELDS.length.times do |index|
+              expect(assignment.public_send(
+                       Assignment::DEFAULT_FIELDS[index]
+                     )).to eq(desired_attributes[index])
+            end
+          end
+        end
+      end
+      context 'when there are multiple assignments' do
+        context 'when some rows of the csv are valid and others are invalid' do
+          before(:each) do
+            # creating 2 rows only containing invalid short identifiers
+            @csv =
```

Comment:
Similar to above, you should define these variables using `let!` instead of `before`